src/PrMerger.js

```diff
@@ -24,12 +24,22 @@ class PrMerger {
         return labels.find(lbl => lbl.name === Config.clearedForMergeLabel()) !== undefined;
     }
 
-    async _getPRList() {
+    async _getPRList(finalizer) {
         let prList = await GH.getPRList();
-        for (let pr of prList)
+        for (let pr of prList) {
             pr.clearedForMerge = await this._clearedForMerge(pr.number);
+            if (finalizer !== null && finalizer.number() === pr.number) {
+                pr.isCurrent = true;
+                pr.processor = finalizer;
+            } else {
+                pr.isCurrent = false;
+                pr.processor = new MergeInitiator(pr);
+            }
+        }
 
-        prList.sort((pr1, pr2) => { return pr2.clearedForMerge - pr1.clearedForMerge || pr1.number - pr2.number; });
+        prList.sort((pr1, pr2) => { return (Config.guardedRun() && (pr2.clearedForMerge - pr1.clearedForMerge)) ||
+                pr2.isCurrent - pr1.isCurrent || pr1.number - pr2.number; });
```

Comment:
I would move the third (pr.number) condition to its own line, to emphasize that we have three sorting criteria.